### PR TITLE
[Storage] Add `x-ms-creation-time` to response on download

### PR DESCRIPTION
**Pull request by vincenttran-msft**
This PR simply generates on the new swagger that will allow for `x-ms-creation-time` to be properly parsed from the response headers and populate the properties.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/azure/storage/blob/_generated/operations/_blob_operations.py [hunk 1]
```diff
@@ -1603,6 +1624,9 @@ def download(
         response_headers = {}
         if response.status_code == 200:
             response_headers["Last-Modified"] = self._deserialize("rfc-1123", response.headers.get("Last-Modified"))
+            response_headers["x-ms-creation-time"] = self._deserialize(
+                "rfc-1123", response.headers.get("x-ms-creation-time")
+            )
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_generated/operations/_blob_operations.py:1629]: Support for 200 status code (OK)

File: sdk/storage/azure-storage-blob/azure/storage/blob/_generated/operations/_blob_operations.py [hunk 2]
```diff
@@ -1681,6 +1705,9 @@ def download(
 
         if response.status_code == 206:
             response_headers["Last-Modified"] = self._deserialize("rfc-1123", response.headers.get("Last-Modified"))
+            response_headers["x-ms-creation-time"] = self._deserialize(
+                "rfc-1123", response.headers.get("x-ms-creation-time")
+            )
```
Comment (vincenttran-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_generated/operations/_blob_operations.py:1710]: Support for 206 status code (partial success i.e. range download)

**Review by jalauzon-msft — COMMENTED** [review]
Hmm I'm hesitant for us to take all the additional changes from the updated autorest version directly into `main` with the upcoming GA release. It may be better for us to try and re-generate with whichever version of autorest was previously used to limit the changes to only the new stuff for creation-time. You can force an autorest version when generating the code but we'll just need to figure out which version.

Also, maybe we should add a test? Or find a test to add a check for the property to?

**Comment by vincenttran-msft** [comment]
> Hmm I'm hesitant for us to take all the additional changes from the updated autorest version directly into `main` with the upcoming GA release. It may be better for us to try and re-generate with whichever version of autorest was previously used to limit the changes to only the new stuff for creation-time. You can force an autorest version when generating the code but we'll just need to figure out which version.
> 
> Also, maybe we should add a test? Or find a test to add a check for the property to?

Good point. I was not aware that the updated autorest version also got slapped in there, but I definitely agree we should drop the autorest version to only introduce the intended change (`x-ms-creation-time` header on response). Let's circle back to this later 😄

**Review by jalauzon-msft — APPROVED** [review]

**Pull request merged**